### PR TITLE
reading csproj for finding out if project is a web project

### PR DESCRIPTION
**Pull request by hiyadav**
you can use these changes if need to take the bug : 10469 further

### CODE REVIEW

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 1]
```diff
@@ -318,29 +319,62 @@ export class dotNetExe {
...
+            resolvedProjectFiles = projectFiles.filter(function (file, index, files): boolean {
+                const directory = path.dirname(file);
+                return tl.exist(path.join(directory, 'web.config'))
+                    || tl.exist(path.join(directory, 'wwwroot'));
             });
 
-            if (!projectFiles.length) {
-                tl.error(tl.loc("noWebProjctFound"));
+            if (!resolvedProjectFiles.length) {
+                projectFiles.forEach((projectFile) => {
```
Comment (sachinma) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: This is wrong. You should use filter to get array. Right now you are just assigning the last element to match to resolvedProjectFilers

Comment (sachinma) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: Add UT for this

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 2]
```diff
@@ -318,29 +319,62 @@ export class dotNetExe {
...
             }
         }
 
         return projectFiles;
     }
 
+    private isWebSdkUsed(projectfile: string) {
+        var fileBuffer: Buffer = fs.readFileSync(projectfile);
+        var fileEncodeType = fileEncoding.detectFileEncoding(projectfile, fileBuffer);
+        var webConfigContent: string = fileBuffer.toString(fileEncodeType[0]);
```
Comment (sachinma) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: will there always be elements at 0th index

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 3]
```diff
@@ -318,29 +319,62 @@ export class dotNetExe {
...
+        if(fileEncodeType[1]) {
+            webConfigContent = webConfigContent.slice(1);
+        }
+        var xmlDocument;
+        var paramFileReplacableValues = {};
+        try {
+            var ltxDomUtiltiyInstance = new ltxdomutility.LtxDomUtility(webConfigContent);
+            xmlDocument = ltxDomUtiltiyInstance.getXmlDom();
+            for(var xmlChildNode of xmlDocument.children) {
+                if(!varUtility.isObject(xmlChildNode)) {
```
Comment (sachinma) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: what is varUtility

File: Tasks/DotNetCoreCLIV2/dotnetcore.ts [hunk 4]
```diff
@@ -318,29 +319,62 @@ export class dotNetExe {
...
+                    continue;
+                }
+                if(xmlChildNode.name == "project" && !!xmlChildNode.attrs['sdk']) {
+                    if (xmlChildNode.attrs['sdk'] == ""
```
Comment (sachinma) [inline Tasks/DotNetCoreCLIV2/dotnetcore.ts]: is this incomplete code?

Comment (hiyadav) [reply Tasks/DotNetCoreCLIV2/dotnetcore.ts]: @sachinma 
The bug is being taken care by Nitin now, i was only transferring the work done so far to him.
Nitin will take it further and raise proper PR.

**Pull request closed without merging**